packages/medusa/src/controllers/products/admin-list-products.ts

```diff
@@ -67,11 +68,20 @@ const listAndCount = async (
     orderBy
   )
 
-  const [products, count] = await productService.listAndCount(
+  const [rawProducts, count] = await productService.listAndCount(
```

Comment:
issue: should be wrapped into a transaction as suggested above

Reply:
comment: since there is multiple action we don't want the data to be updated during the flow